lib/action_policy/rspec/be_authorized_to.rb

```diff
@@ -32,10 +32,16 @@ def with(policy)
         self
       end
 
+      def with_context(context)
+        @context = context
+        self
+      end
+
       def match(_expected, actual)
         raise "This matcher only supports block expectations" unless actual.is_a?(Proc)
 
         @policy ||= ::ActionPolicy.lookup(target)
+        @context ||= {}
```

Comment:
What about using `nil` by default? That would allow us to conditionally include the context to the message.

I'm worried about potentially too verbose error messages if we add `context` (or `authorization_context`) by default

Reply:
Adding the context default as nil would cause the check to fail when `@context` was nil.
https://github.com/palkan/action_policy/blob/e78c156c473728c9e4f85644b7e3de0dc157e2dc/lib/action_policy/testing.rb#L16-L20

Found easier to stick with context as default as an empty hash and add the context message based on that.